### PR TITLE
[Fix #47] Remove flatland/useful as a dep

### LINKED ISSUE #47: Ordered pulls in useful
which in turn pulls in a bunch of other deps.

### PR DESCRIPTION
**Pull request by slipset**
- Remove flatland/useful as a dep
- Stop using `delegating-deftype`

### CODE REVIEW

File: src/flatland/ordered/map.clj [hunk 1]
```diff
@@ -93,6 +76,26 @@
     (hasheq-ordered-map this))
 
   IPersistentMap
+  (equiv [this other]
```
Comment (danielcompton) [inline src/flatland/ordered/map.clj]: These are just copied from the underlying implementation right?

**Comment by vxe** [comment]
any updates?  looks like this a blocker for this https://github.com/borkdude/babashka/issues/263

**Comment by slipset** [comment]
I don't have permission to publish a new artefact to clojars as this publishes to org.flatland.
Hopefully @danielcompton can do this.

**Comment by danielcompton** [comment]
Yep just deployed this now, thanks!

**Pull request merged**